Reservaciones/Screens/Login.js

```diff
@@ -1,44 +1,271 @@
+  return auth().signInWithCredential(googleCredential);
+}
 
+function LoginApp() {
+  const [initializing, setInitializing] = useState(true);
+  const [user, setUser] = useState();
 
-const Login = () => {
-   
-   return (
+  function onAuthStateChanged(user) {
+    setUser(user);
+    if (initializing) setInitializing(false);
+  }
+
+  useEffect(() => {
+    const subscriber = auth().onAuthStateChanged(onAuthStateChanged);
+    return subscriber;
+  }, []);
+
+  if (initializing) return null;
+
+  if (!user) {
+    return (
+      <View>
+        <Text>Login principal</Text>
+      </View>
+    );
+  }
+
+  return (
+    <View>
+      <Text>Bienvenido {user.email}</Text>
+    </View>
+  );
+}
+
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+      isPasswordHidden: true,
+      userEmail: '',
+      userPass: '',
+      errorLogin: '',
+    };
+  }
+
+  signOut = () => {
+    auth()
+      .signOut()
+      .then(() => console.log('User signed out!'));
+  };
+
+  EmailPassSignIn = () => {
+    auth()
+      .createUserWithEmailAndPassword(
+        'jhon.doe@example.com',
+        'SupesssrSecretPassword!',
+      )
+      .then(() => {
+        console.log('User account created & signed in!');
+      })
+      .catch((error) => {
+        if (error.code === 'auth/email-already-in-use') {
+          console.log('That email address is already in use!');
+        }
+        if (error.code === 'auth/invalid-email') {
+          console.log('That email address is invalid!');
+        }
+        console.error(error);
+      });
+  };
+
+  handlerFocus = (input) => {
+    this.setState({
+      [input]: true,
+    });
+  };
+  handlerBlur = (input) => {
+    let leghtEmail = this.state.userEmail;
+    if (leghtEmail.trim() === '') {
+      this.setState({
+        [input]: false,
+      });
+    } else {
+      this.setState({
+        [input]: true,
+      });
+    }
+  };
 
+  changeShowPass = () => {
+    const {isPasswordHidden} = this.state;
+    this.setState({isPasswordHidden: !isPasswordHidden});
+  };
+
+  async SignInWithEmailPass(email, password) {
+    try {
+      if (email.trim() === '') {
+        alert('Indtroduzca un email por favor');
+      } else if (password.trim() === '') {
+        alert('Indtroduzca un password por favor');
+      } else {
+       
+        this.setState({
+          loading: true,
+          errorLogin: ''
+        });
+
+        await auth().signInWithEmailAndPassword(email, password);
+        // alert('entro por email y pass');
+        //await auth().onAuthStateChanged(user => {  alert(user.email);  })
+         
+        setTimeout(() => {
+          this.setState({
+            loading: false,
+          });
+        }, 2500);
+      }
+    } catch (error) {
+      let errorMessage = error.message.toString(error);
+      this.setState({errorLogin: errorMessage});
```

Comment:
use constants if the value is not modified
```suggestion
      const errorMessage = error.message.toString(error);
      this.setState({errorLogin: errorMessage});
```